### PR TITLE
Fix building with gcc 12

### PR DESCRIPTION
**Pull request by thomasspriggs**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by TGWDB** [comment]
Looks like a good alternative to #7540 , I notice the new job has significantly higher runtime than the straight gcc one. This partly due to me wanting to see the typical build time in the CI job comment as in the other jobs (obviously not really known until tested).

**Comment by thomasspriggs** [comment]
> Looks like a good alternative to #7540 , I notice the new job has significantly higher runtime than the straight gcc one. This partly due to me wanting to see the typical build time in the CI job comment as in the other jobs (obviously not really known until tested).

I'd guess the longer run time will be due to the build cache being cold. I'll restart the jobs to get a better idea of the variation.

**Comment by thomasspriggs** [comment]
For my own reference the total run time for the initial run was 46m 26s, with 23m 44s of that being the build time.

File: .github/workflows/pull-request-checks.yaml [hunk 1]
```diff
@@ -365,6 +365,58 @@ jobs:
...
+        run: |
+          wget -O cvc5 https://github.com/cvc5/cvc5/releases/download/cvc5-${{env.cvc5-version}}/cvc5-Linux
+          chmod u+x cvc5
+          mv cvc5 /usr/local/bin
+          cvc5 --version
+      - name: Prepare ccache
+        uses: actions/cache@v3
+        with:
+          path: .ccache
+          key: ${{ runner.os }}-22.04-Release-${{ github.ref }}-${{ github.sha }}-PR
```
Comment (tautschnig) [inline .github/workflows/pull-request-checks.yaml]: This also needs a "gcc-12-" after "Release"

**Review by tautschnig — APPROVED** [review]

File: .github/workflows/pull-request-checks.yaml (hunk 1 above)
Comment (thomasspriggs) [reply .github/workflows/pull-request-checks.yaml]: Added.

Comment (tautschnig) [reply .github/workflows/pull-request-checks.yaml]: This, by the way, might explain why the build cache remained cold...

Comment (thomasspriggs) [reply .github/workflows/pull-request-checks.yaml]: It might indeed.

**Review by TGWDB — APPROVED** [review]
Assuming the cache/timing comment can be resolved.

**Comment by thomasspriggs** [comment]
Note to self - job time is 45m 52s, 23m 28s building after the cache fix, with cold cache.

**Review by martin-cs — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Comment by thomasspriggs** [comment]
Note to self - Job time is 26m 43s, of which 38s is building, when cache is warm.

**Pull request merged**